xblockutils/public/studio_container.js

```diff
@@ -1,20 +1,21 @@
 function StudioContainerXBlockWithNestedXBlocksMixin(runtime, element) {
     var $buttons = $(".add-xblock-component-button", element),
+        $view = $('.add-xblock-component', element),
         $element = $(element);
 
     function isSingleInstance($button) {
         return $button.data('single-instance');
     }
 
-    $buttons.click(function(ev) {
-        var $this = $(this);
-        if ($this.is('.disabled')) {
+    $view.on('click', '.add-xblock-component-button', function(ev) {
```

Comment:
@eliangcs Could you please add a comment explaining why we need to use delegated events here? This wouldn't be obvious to someone looking at this code later on/outside of the context of this PR.